### DISCUSSION TITLE
리액트 assignment1 기완님 예시답안 jsx

### DISCUSSION THREAD

**Discussion by Arsture** [wafflestudio/seminar-2021#560] [Q&A] [answered]
## 요약

기완님께서 만들어주신 예시답안에  컴포넌트 파일확장자가 jsx인데 파일확장자 js랑 기능적으로 어떤 차이가 있는지 궁금합니다!

<br><br><br><br>

## 상황

예시답안을 보다 궁금해졌습니다! 아래 캡쳐에서 보실 수 있듯이 jsx 파일확장자도 js와 문법이 똑같은 것 같아서(`import ~~`, `const Divider = () => ~~`, `export ~~`) 어떤 점이 다른지 궁금합니다.

<br><br><br><br>

## 문제 내용

![image](https://user-images.githubusercontent.com/87017787/135195023-aa391988-1447-4b22-8bd6-6a31a2c1b9f9.png)
![image](https://user-images.githubusercontent.com/87017787/135195053-23854f20-5c84-4a2b-99b2-7c9db40d56b3.png)

**Comment by ars-ki-00** [comment, marked as answer]
엄밀히 따지고 보면, 아무런 차이도 없습니다.
리액트는 js의 라이브러리 중 하나일 뿐이므로, 어떠한 확장자를 사용하더라도 동일한 결과를 냅니다.

단, 클린 코드를 지향하기 때문에, plain javascript 가 아닌 jsx 는 `.jsx` 파일에 넣어 놨을 뿐이에요. 
확장자만 보고도 이 파일이 컴포넌트인지 util인지를 구분하기 위함이었습니다.

**Comment by Arsture** [comment]
아하 넵 감사합니당!
